pkg/http/server_test.go

```diff
@@ -5,22 +5,148 @@ import (
 	"errors"
 	"fmt"
 	"io"
+	"net"
 	"net/http"
 	"os"
+	"os/exec"
 
 	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/apis/options"
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/ginkgo/extensions/table"
 	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gbytes"
+	. "github.com/onsi/gomega/gexec"
+	. "github.com/onsi/gomega/gleak"
 )
 
 const hello = "Hello World!"
 
 var _ = Describe("Server", func() {
+
+	Context("StandaloneServer", func() {
+		type startServerTableInput struct {
+			fdListener       bool
+			httpAddress      string
+			listenAddress    string
+			protocol         string
+			expectStartupErr string
+			skipEnv          bool
+		}
+
+		var ctx context.Context
+		var cancel context.CancelFunc
+
+		BeforeEach(func() {
+			ctx, cancel = context.WithCancel(context.Background())
+		})
+
+		AfterEach(func() {
+			cancel()
+		})
+
+		args := []string{
+			"--cookie-secret=1234123412341234",
+			"--client-id=client_id",
+			"--client-secret=client_secret",
+			"--email-domain=*",
+		}
+
+		DescribeTable("When starting the new server from the options", func(in *startServerTableInput) {
+			var listenAddr string
+			cmdArgs := args
+			if in.httpAddress != "" {
+				cmdArgs = append(cmdArgs, fmt.Sprintf("--http-address=%s", in.httpAddress))
+			}
+
+			c := exec.CommandContext(ctx, cmdPath, cmdArgs...)
+			if in.fdListener {
+				c.ExtraFiles = []*os.File{}
+				if in.listenAddress != "" {
+					l, err := net.Listen(in.protocol, in.listenAddress)
+					Expect(err).ToNot(HaveOccurred())
+					listenAddr = fmt.Sprintf("http://%s/", l.Addr().String())
+
+					var f *os.File
+					switch ln := l.(type) {
+					case *net.TCPListener:
+						f, err = ln.File()
+					}
+					Expect(err).ToNot(HaveOccurred())
+
+					c.ExtraFiles = append(c.ExtraFiles, f)
+				}
+
+				c.Env = []string{
+					fmt.Sprintf("LISTEN_FDS=%d", len(c.ExtraFiles)),
+				}
+				if !in.skipEnv {
+					c.Env = append(c.Env, "FIX_LISTEN_PID=1")
+				}
+			}
+			session, err := Start(c, GinkgoWriter, GinkgoWriter)
+			Expect(err).ToNot(HaveOccurred())
+
+			Eventually(session).Should(Say(".*OAuthProxy configured.*"))
+			Eventually(session).Should(Say(".*Cookie settings:.*"))
+
+			if in.expectStartupErr != "" {
+				Eventually(session.Err).Should(Say(in.expectStartupErr))
+			} else {
+
+				resp, err := httpGet(ctx, listenAddr)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(resp.StatusCode).To(Equal(403))
+				Eventually(session).Should(Say(".*No valid authentication in request. Initiating login."))
+			}
+		},
+			Entry("with an fd valid http IPv4 bind address", &startServerTableInput{
+				fdListener:    true,
+				httpAddress:   "FD:3",
```

Comment:
Is this case sensitive?

Reply:
It is.

Reply:
https://github.com/isodude/oauth2-proxy/blob/9c9e10fda9731e29ac5baf642f9edae4c5464555/pkg/http/server.go#L110

Reply:
The line you link links to it being case **insensitive**, I don't mind either way but that should be documented and tests should cover both sides of the story

Reply:
Hm, I did add a test with "Fd" as an example. But maybe adding that to the docs is a good idea.

Reply:
Added in the docs as well.